sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1909,458 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class QuantileNormalizer(BaseEstimator, TransformerMixin):
```

Comment:
I'm still not happy to call this normalization when we usually consider normalization to be row-wise and this is column-wise. I think newcomers are already confused enough about row-normed vs standardized features.

I would rather call this a scaler even if its transformation is non-linear. I think users care more that similar objects have similar names and APIs.